### PR TITLE
node: poller timeout

### PR DESCRIPTION
**Pull request by bruce-riley**
The provider we are using for Arbitrum testnet limits the number of queries we can do. This was causing the poller connector to hang while calling eth_getBlockByNumber(). This PR adds a timeout to those calls so that the runnable will die and get restarted.

Change-Id: Ia324f1ac482fa9c5bea2b501970f0b22b16e67ce

### CODE REVIEW

File: node/pkg/watchers/evm/connectors/poller.go [hunk 1]
```diff
@@ -74,12 +74,15 @@ func (b *BlockPollConnector) run(ctx context.Context) error {
 }
 
 func (b *BlockPollConnector) pollBlocks(ctx context.Context, logger *zap.Logger, lastBlock *NewBlock) (lastPublishedBlock *NewBlock, retErr error) {
+	timeout, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
```
Comment (jynnantonix) [inline node/pkg/watchers/evm/connectors/poller.go:78]: Are we concerned about any state that gets lost when we time out and the whole runnable is killed?  Would it be possible to rate limit on our end instead?  For example, we could buffer requests in a channel and have a separate goroutine send them out in a rate-limited manner.

File: node/pkg/watchers/evm/watcher.go [hunk 2]
```diff
@@ -634,6 +634,7 @@ func (w *Watcher) Run(ctx context.Context) error {
 		}
 	}()
 
+	readiness.SetReady(w.readiness)
```
Comment (jynnantonix) [inline node/pkg/watchers/evm/watcher.go:637]: Unrelated change?

**Review by jynnantonix — COMMENTED** [review]
It would be nice to put the PR description in the commit as well so that it shows up in the git history.

File: node/pkg/watchers/evm/watcher.go (hunk 2 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go:637]: Unrelated but related. When it can take a long time to reach finality (such as Arbitrum in testnet), it takes a long time for the runable to be marked ready, which causes tilt initialization to hang. This lets initialization continue sooner.

Comment (jynnantonix) [reply node/pkg/watchers/evm/watcher.go:637]: Ok, then please add a comment here explaining that.

File: node/pkg/watchers/evm/connectors/poller.go (hunk 1 above)
Comment (hendrikhofstadt) [reply node/pkg/watchers/evm/connectors/poller.go:78]: Agree, maybe we should be more graceful in L69 like the original version of the poller (pre my refactoring PR), allow X timeouts with some kind of backoff. This would automatically retry blocks that were missed.

Comment (bruce-riley) [reply node/pkg/watchers/evm/connectors/poller.go:78]: This change is being made to address an issue with Arbitrum in testnet because the provider we are using keeps disconnecting us. This should not happen in mainnet, and if it does, I feel it is reasonable for the runable to die. If we see the runable die in mainnet, we need to address the underlying cause, rather than preemptively code around the issue.

File: node/pkg/watchers/evm/watcher.go (hunk 2 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go:639]: Added a comment.

File: node/pkg/watchers/evm/connectors/poller.go (hunk 1 above)
Comment (jynnantonix) [reply node/pkg/watchers/evm/connectors/poller.go:78]: Ack.  I'm fine with leaving this as is and we can revisit if it becomes a problem in the future but please add a comment here explaining the situation for future readers who don't have / forgot the context.

**Review by jynnantonix — APPROVED** [review]

File: node/pkg/watchers/evm/connectors/poller.go (hunk 1 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/connectors/poller.go:82]: Added a comment.

**Comment by bruce-riley** [comment]
Added a couple of retries. Seems to help with Arbitrum testnet.

**Review by evan-gray — APPROVED** [review]

**Review by jynnantonix — APPROVED** [review]
